### PR TITLE
Add proof summary scripts (work in progress)

### PR DESCRIPTION
**Pull request by markrtuttle**
This directory contains a script that scans the viewer output for a set of and produce a summary of the stubs and undefined functions being used in the proofs.  This is a work-in-progress, and will ultimately be incorporated into viewer and litani.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: summary/summary.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+            result = summarize_project(project_path, proofs_path)
+            if result:
+                summary[project] = result
+
+    if args.summarize:
+        print(json.dumps(summary, indent=2))
+
+    if args.chart:
+#        chart_projects_csv(summary)
+        print(chart_projects_json(summary))
```
Comment (feliperodri) [inline summary/summary.py:253]: why print?

File: summary/projects.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  "PKCS11": {
+    "github": "git@github.com:FreeRTOS/corePKCS11",
+    "proofs": "test/cbmc/proofs"
+  },
+  "SNTP": {
+    "github": "git@github.com:FreeRTOS/coreSNTP",
+    "proofs": "test/cbmc/proofs"
+  }
+}
```
[the remark this answers is not in the document]
Comment (markrtuttle) [reply summary/projects.json:58]: The keys like "Shadow" are abbreviations for the repository names like "device-shadow-for-aws-iot-embedded-sdk" that are used as column headers in the table produced, so templates are not a complete replacement.  Templates are also a brittle solution when the starter kit makes not requirements about where the cbmc root is placed in the repository.  The real problem is that this json file is project specific and doesn't belong in a tool distribution directory.  Let's revisit this when we have merged this code into viewer itself.

File: summary/summary.py (hunk 1 above)
Comment (markrtuttle) [reply summary/summary.py:253]: The semantics of the --summarize and --chart flags are to produce a summary or a chart.  How do you recommend producing them?

File: summary/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      "Baseball": {
+        "github": "https://github.com/johnqpublic/baseball-fantasy-team.git",
+        "proofs": "tools/cbmc/proofs"
+      },
+      "Football": {
+        "github": "https://github.com/johnqpublic/football-fantasy-team.git",
+        "proofs": "test/cbmc/proofs"
+      }
+    }
+```
```
Comment (feliperodri) [inline summary/README.md:30]: Nice!

**Pull request merged**